### PR TITLE
deposit: oai syncing cli

### PR DESCRIPTION
**Pull request by krzysztof**
Signed-off-by: Krzysztof Nowak <k.nowak@cern.ch>

### CODE REVIEW

File: zenodo/modules/utils/cli.py [hunk 1]
```diff
@@ -225,3 +228,21 @@ def list_files(recid):
...
+
+
+@utils.command('sync_oai')
+@click.option('--eager', '-e', is_flag=True, default=False)
+@with_appcontext
+def sync_oai(eager):
+    """Update OAI IDs in the records."""
+    pids = PersistentIdentifier.query.filter(
+        PersistentIdentifier.pid_type == 'recid',
+        PersistentIdentifier.object_uuid is not None,
```
Comment (lnielsen) [inline zenodo/modules/utils/cli.py]: + PersistentIdentifier.object_type == 'rec',
- PersistentIdentifier.object_uuid is not None,

File: zenodo/modules/utils/tasks.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                'pointing to a different object ({id2})'.format(
+                    pid=pid, id=uuid, id2=str(pid.get_assigned_object()),
+                    recid=recid_s))
+    except NoResultFound:
+        zenodo_oaiid_minter(rec.id, rec)
+        synced_sets = get_synced_sets(rec)
+        rec['_oai']['sets'] = synced_sets
+        rec.commit()
+        db.session.commit()
+    except MultipleResultsFound:
```
Comment (lnielsen) [inline zenodo/modules/utils/tasks.py]: can be removed

File: zenodo/modules/utils/tasks.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+from zenodo.modules.records.minters import zenodo_oaiid_minter
+
+logger = get_task_logger(__name__)
+
+
+def get_synced_sets(record):
+    """Get the synced OAI set list."""
+    comms = record.get('communities', [])
+    sets = record.get('_oai', {}).get('sets', [])
+    other_sets = [s for s in sets if not s.startswith('user-')]
```
Comment (krzysztof) [inline zenodo/modules/utils/tasks.py]: Should also check if query is defined

File: zenodo/modules/utils/tasks.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+def comm_sets_match(record):
+    """Check if communities definition and OAI sets match."""
+    # OAI sets of communities
+    comms = record.get('communities', [])
+    s_comms = ['user-{0}'.format(c) for c in comms]
+
+    # Community-based OAI sets
+    sets = record.get('_oai', {}).get('sets', [])
+    c_sets = [s for s in sets if s.startswith('user-')]
```
Comment (krzysztof) [inline zenodo/modules/utils/tasks.py]: Should also check if query is defined

**Comment by krzysztof** [comment]
@lnielsen Can you take another look?

**Comment by lnielsen** [comment]
I've tried to run it on a prod dump, but I'm still getting inconsistent results from OAI-PMH so there might be more at play here. We can talk IRL about it tomorrow.

**Comment by lnielsen** [comment]
Sorry, my fault :-) Forgot to run the bulk indexing :-)

**Pull request merged**